python-checks/src/test/java/org/sonar/python/checks/UnusedLocalVariableCheckTest.java

```diff
@@ -42,178 +46,250 @@ void custom() {
+        "tuple1",
+        """
+          def using_tuples():
+            x, y = (1, 2)
+            print x
+          """,
+        """
+          def using_tuples():
+            x, _ = (1, 2)
+            print x 
+          """,
+        "Replace with \"_\""),
+      new QuickFixTestData(
+        "tuple2",
+        """
+          def using_tuples():
+              x, y = (1, 2)
+              y = 5
+              print x 
+          """,
+        """
+          def using_tuples():
+              x, _ = (1, 2)
+              y = 5
+              print x 
+          """,
+        "Replace with \"_\""
+      ),
+      new QuickFixTestData(
+        "exceptClause",
+        """
+          def foo():
+            try:
+              ...
+            except Type as e:
+              ...
+          """,
+        """
+          def foo():
+            try:
+              ...
+            except Type:
+              ...
+          """,
+        "Remove the unused local variable"
+      ),
+      new QuickFixTestData(
+        "loopIndex",
+        """
+          def loop_index():
+            for i in range(10):
+              print("Hello")
+          """,
+        """
+          def loop_index():
+            for _ in range(10):
+              print("Hello")
+          """,
+        "Replace with \"_\""
+      ),
+      new QuickFixTestData(
+        "loopIndexComprehension",
+        """
+          def loop_index():
+           return [True for i in range(10)]
+          """,
+        """
+          def loop_index():
+           return [True for _ in range(10)]
+          """,
+        "Replace with \"_\""
+      ),
+      new QuickFixTestData(
+        "loopIndexComprehensionClass",
+        """
+          class A():
+            _ = True
+            def __init__(self):
+              for i in range(5):
+                print("print")
+          """,
+        """
+          class A():
+            _ = True
+            def __init__(self):
+              for _ in range(5):
+                print("print")
+          """,
+        "Replace with \"_\""
+      ),
+      new QuickFixTestData(
+        "assignment",
+        """
+          def foo():
+            x = bar()
+            y = True
+            return y 
+          """,
+        """
+          def foo():
+            bar()
+            y = True
+            return y 
+          """,
+        "Remove assignment target"
+      ),
+      new QuickFixTestData(
+        "assignmentExpression1",
+        """
+          def foo():
+            if any((i := j) % 2 == 1 for j in range(3)):
+              return 
+          """,
+        """
+          def foo():
+            if any(j % 2 == 1 for j in range(3)):
+              return 
+          """,
+        "Remove assignment target"
+      ),
+      new QuickFixTestData("assignmentExpression2",
+        """
+          def foo():
+            if any((i := j*2) % 2 == 1 for j in range(3)):
+              return
+          """,
```

Comment:
We could maybe check that the quick fix does not break anything even with set/dict comprehension.